tests/unit/test_query.py

```diff
@@ -1019,6 +1058,234 @@ def test_query_errors():
+            default_options=options,
+            orders=[],
+        )
+        q2 = q.bind()
+        assert q2.kind == "Foo"
+
+    @staticmethod
+    @pytest.mark.usefixtures("in_context")
+    def test_bind_with_parameter_ancestor():
+        options = query.QueryOptions(kind="Bar")
+        q = query.Query(
+            kind="Foo",
+            ancestor=query.Parameter("xyz"),
+            namespace="space",
+            app="app",
+            group_by=["X"],
+            projection=[model.Property(name="x")],
+            filters=query.FilterNode("f", None, None),
+            default_options=options,
+            orders=[],
+        )
+        key = key_module.Key("a", "b", app="app", namespace="space")
+        q2 = q.bind(xyz=key)
+        assert q2.kind == "Foo"
+
+    @staticmethod
+    @pytest.mark.usefixtures("in_context")
+    def test_bind_with_bound_and_unbound():
+        options = query.QueryOptions(kind="Bar")
+        q = query.Query(
+            kind="Foo",
+            ancestor=query.Parameter("xyz"),
+            namespace="space",
+            app="app",
+            group_by=["X"],
+            projection=[model.Property(name="x")],
+            filters=query.FilterNode("f", None, None),
+            default_options=options,
+            orders=[],
+        )
+        with pytest.raises(exceptions.BadArgumentError):
+            q.bind(42, "xyz", xyz="1")
+
+    @staticmethod
+    @pytest.mark.usefixtures("in_context")
+    def test_bind_error():
+        q = query.Query()
+        with pytest.raises(exceptions.BadArgumentError):
+            q.bind(42)
+
+    @staticmethod
+    @pytest.mark.usefixtures("in_context")
+    def test__get_query(context):
+        options = query.QueryOptions(kind="Bar")
+        q = query.Query(
+            kind="Foo",
+            ancestor=key_module.Key("a", "b", app="app", namespace="space"),
+            namespace="space",
+            app="app",
+            group_by=["X"],
+            projection=[model.Property(name="x")],
+            filters=query.FilterNode("f", None, None),
+            default_options=options,
+            orders=[],
+        )
+        dsq = q._get_query(context.client)
+        assert isinstance(dsq, datastore.Query)
+
+    @staticmethod
+    @pytest.mark.usefixtures("in_context")
+    def test__get_query_with_conjunction_node(context):
+        options = query.QueryOptions(kind="Bar")
+        filters = [
+            query.FilterNode("y", ">", 0),
+            query.FilterNode("y", "<", 1000),
+        ]
+        q = query.Query(
+            kind="Foo",
+            ancestor=key_module.Key("a", "b", app="app", namespace="space"),
+            namespace="space",
+            app="app",
+            group_by=["X"],
+            projection=[model.Property(name="x")],
+            filters=query.ConjunctionNode(*filters),
+            default_options=options,
+            orders=[],
+        )
+        dsq = q._get_query(context.client)
```

Comment:
https://github.com/google/styleguide/blob/gh-pages/pyguide.md#316-naming

Reply:
By the time I saw your commit regarding naming on PR #44 this was already done, but I will fix all of these, thanks.